plugins/pulp_docker/plugins/registry.py

```diff
@@ -368,8 +368,39 @@ def get_manifest(self, reference):
         :return:          A 2-tuple of the digest and the manifest, both basestrings
         :rtype:           tuple
         """
+        manifests = []
+        request_headers = {}
+        content_type_header = 'content-type'
+        schema1 = 'application/vnd.docker.distribution.manifest.v1+json'
+        schema2 = 'application/vnd.docker.distribution.manifest.v2+json'
         path = self.MANIFEST_PATH.format(name=self.name, reference=reference)
-        headers, manifest = self._get_path(path)
+        # set the headers for first request
+        request_headers['Accept'] = schema2
+        headers, manifest = self._get_path(path, headers=request_headers)
+        digest = self._digest_check(headers, manifest)
+
+        # add manifest and digest
+        manifests.append((manifest, digest))
+
+        # we intentionally first asked schema2, because it might happen that
+        # registry will have just schema1 and even if in request headers we
+        # set schema2, registry will anyway return schema1
+        # this way we will not make unecessary separate second request for schema1
+        if headers[content_type_header] == schema2:
+            request_headers['Accept'] = schema1
+            headers, manifest = self._get_path(path, headers=request_headers)
+            digest = self._digest_check(headers, manifest)
+
+            # add manifest and digest
+            manifests.append((manifest, digest))
+
+        # returned list will be whether:
+        # [(S2, digest), (S1, digest)]
+        # or
+        # [(S1, digest)]
+        return manifests
```

Comment:
whether having a defined structure or not, we would anyway need to do inspection if not on data, then on dict keys.
i did some changes where we do not depend anymore on the return structure and let me now if this approach is acceptable.